src/main/java/org/eurekaclinical/user/service/util/AbstractOAuthRegistrationUserResource.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * #%L
+ * Eureka! Clinical User Webapp
+ * %%
+ * Copyright (C) 2016 Emory University
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package org.eurekaclinical.user.service.util;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+
+import org.scribe.up.provider.OAuthProvider;
+
+import org.eurekaclinical.scribeupext.profile.EurekaProfile;
+import org.eurekaclinical.standardapis.exception.HttpStatusException;
+import org.scribe.up.credential.OAuthCredential;
+import org.scribe.up.session.HttpUserSession;
+
+/**
+ *
+ * @author miaoai
+ */
+public class AbstractOAuthRegistrationUserResource<E extends EurekaProfile> {
+
+    private final OAuthProvider provider;
+    
+    
+
+    public AbstractOAuthRegistrationUserResource(OAuthProvider oauthProvider) {
+        assert oauthProvider != null : "provider cannot be null";
+        this.provider = oauthProvider;
+    }
+    
+    
+    
+    public E getEurekaProfile(HttpServletRequest req) throws HttpStatusException {
+        assert req != null : "request cannot be null";
+        OAuthCredential credential
+                = this.provider.getCredential(
+                        new HttpUserSession(req.getSession()),
```

Comment:
Please confirm that we're not storing any user information in the session as a result of the OAuth workflow.